docs/encryption.md

```diff
@@ -0,0 +1,118 @@
+   1   sha256:383a3d4c6789667dbfb6b3742492c4a925315e750f99a5d664ff72f2bb0ae659        linux/386       191   jwe,pkcs7    [jwe], [pkcs7]
+   0   sha256:8aea19b10fd75004ab8fd2d02df719c06528ad3539e686a2d26c933d53f25675    linux/ppc64le   2195242
+   1   sha256:965a60ab5513a2eee33f4d960b63ee347215eb31d06a4ed61f6d90d209462d76    linux/ppc64le       193   jwe,pkcs7    [jwe], [pkcs7]
+   0   sha256:783541963cb4e52173193fe947bb7a7f7e5a6657a4cbbb6b8b077bbee7255605      linux/s390x   2307762
+   1   sha256:69d3260b3f5430ade9a3ee0f1b71a32a8e4ef268552beeae29930a8795dc54bf      linux/s390x       192   jwe,pkcs7    [jwe], [pkcs7]
+```
+
+## Decrypt
+
+The following command performs an decryption of the encrypted image `docker.io/library/alpine:enc` to the image tag `docker.io/library/alpine:latest`.
+The decryption is done by passing in the private key that corresponds to at least one of the recipients of the encrypted image.
+
+```
+$ ctr images decrypt \
+    --key /tmp/tmp.AGrSDkaSad/mykey2.pem \
+    docker.io/library/alpine:enc docker.io/library/alpine:dec
+
+Decrypting docker.io/library/alpine:enc to docker.io/library/alpine:dec
+```
+
+# Other Consumers
+
+Other consumers of the encryption routine can include the [containerd diff plugin](https://github.com/containerd/containerd/tree/master/services/diff) to have the decryption be performed in the pull path, as well as the docker CLI or other build tools that wish to provide an encryption option.
+The current draft of exposed interfaces we believe will be used by consumers are as follows:
+
+```
+/* Functions */
+
+// EncryptImage encrypts an image; it accepts either an OCI descriptor representing a manifest list or a single manifest
+func EncryptImage(ctx context.Context, cs content.Store, desc ocispec.Descriptor, cc *encconfig.CryptoConfig, lf *encryption.LayerFilter) (ocispec.Descriptor, bool, error)
+
+// DecryptImage decrypts an image; it accepts either an OCI descriptor representing a manifest list or a single manifest
+func DecryptImage(ctx context.Context, cs content.Store, desc ocispec.Descriptor, cc *encconfig.CryptoConfig, lf *encryption.LayerFilter) (ocispec.Descriptor, bool, error)
+
+// CheckAuthorization checks whether a user has the right keys to be allowed to access an image (every layer)
+// It takes decrypting of the layers only as far as decrypting the asymmetrically encrypted data
+// The decryption is only done for the current platform
+func CheckAuthorization(ctx context.Context, cs content.Store, desc ocispec.Descriptor, dc *encconfig.DecryptConfig) error
+
+// GetImageLayerInfo gets the image key Ids necessary for decrypting an image
+// We determine the KeyIds starting with  the given OCI Decriptor, recursing to lower-level descriptors
+// until we get them from the layer descriptors
+func GetImageLayerInfo(ctx context.Context, cs content.Store, desc ocispec.Descriptor, lf *encryption.LayerFilter, layerIndex int32) ([]encryption.LayerInfo, error)
```

Comment:
Is there a PR with this? This interface doesn't make sense. First off, referencing layers by their index creates a dependency on layer ordering. Image format in the future might not have that model. Making it follow this model will make it hard to change in the future.

Reply:
I removed the layerIndex now from this public function's signature.

Reply:
Updated docs to reflect new change cda9d21a57c541e5b2692f1b0f8d29826fc4daec